wpengine/faustjs
# Create client for App Router

## Tasks


## Description

This PR adds the `getClient` function to the `@faustwp/experimental-app-router` package to be used to fetch data from WordPress in RSCs (React Server Components). Additionally, this PR adds a new example project called `app-router` for testing changes and features to the experimental app router work.



## Related Issue(s):



## Testing

1. Check out the PR
2. Run `npm install` at the monorepo root
3. Build the packages `npm run build` from the monorepo root
4. Create a `.env.local` file from the `.env.local.sample` in `examples/next/app-router`
5. Run `npm run generate -w @faustwp/app-router-example` to generate your site's possible types
6. Run `npm run dev -w @faustwp/app-router-example` to start the example project on http://localhost:3000
7. View the homepage and notice the title/description and menus are being queried. Go to `/[postSlug]` to see post data being queried.



## Screenshots



## Documentation Changes



## Dependant PRs
